### ISSUE TITLE
Calcular o coeficiente de Pearson, coeficiente de Spearman e MSE no processo de predição da rede siamesa

### ISSUE THREAD

**Issue by HaraHeique** [HaraHeique/TCC-rede-neural-siamesa#38]
Segundo as pesquisas do professor @fidelis grande maioria dos artigos científicos tem o problema similar ao do TCC utilizam as seguintes medidas de avaliação do modelo na etapa de predição (testes):

- Coeficiente de Pearson;
- Coeficiente de Spearman;
- MSE.

Logo o ideal é saber como calculá-los usando _Keras/Tensorflow_ e por fim apresentar os resultados.

**Comment by HaraHeique** [comment]
Links importantes relativos aos cálculos das métricas e explicações sobre elas:

#### Coeficientes de Pearson e Spearman
- https://realpython.com/numpy-scipy-pandas-correlation-python/#example-numpy-correlation-calculation
- https://machinelearningmastery.com/how-to-use-correlation-to-understand-the-relationship-between-variables/
- https://towardsdatascience.com/feature-selection-with-pandas-e3690ad8504b

Segundo um dos links o que mais entendi o seguinte tanto para o coeficiente Pearson quanto para o Spearman é o seguinte:
> 
> The correlation coefficient has values between -1 to 1
> - A value closer to 0 implies weaker correlation (exact 0 implying no correlation)
> - A value closer to 1 implies stronger positive correlation
> - A value closer to -1 implies stronger negative correlation

Também que o Pearson é voltado para correlações lineares. Já o Spearman são não lineares.

#### MSE (Mean Squared Error)
- https://scikit-learn.org/stable/modules/generated/sklearn.metrics.mean_squared_error.html
- https://www.tensorflow.org/api_docs/python/tf/keras/metrics/mean_squared_error

Segundo o que pesquisei sobre MSE temos o seguinte

> - The mean squared error (MSE) tells you how close a regression line is to a set of points
> - There is no correct value for MSE. Simply put, the lower the value the better and 0 means the model is perfect.

**Comment by HaraHeique** [comment]
@Fideliszan 

Até então alguns resultados não foram tão satisfatórios, mas realizar os treinamentos utilizando o otimizador _hyperas_. Alguns dos resultados que podem ser visualizados através da planilha criada (implementada para o card #41) são esses:

![image](https://user-images.githubusercontent.com/26990166/128135251-014d5fe3-f75d-4721-bc76-c8c7f854b301.png)

E os resultados da média da predição dos autores para cada uma das linhas da imagem CSV mostrada logo acima são respectivamente:

![image](https://user-images.githubusercontent.com/26990166/128135505-ffc4123a-36da-4927-bc06-6780892e1a68.png)

![image](https://user-images.githubusercontent.com/26990166/128135546-013eff4b-3b90-47e6-9951-dc60ad0023b1.png)

![image](https://user-images.githubusercontent.com/26990166/128135591-763f6c98-2594-4cea-a1c8-7eafcd1fabba.png)

O qual a última imagem foi um teste realizado utilizando os hiperparâmetros já coletados em testes passados no hyperas o qual é:

> Evalutation of best performing model:
> 113/113 [==============================] - 7s 63ms/step - loss: 0.3139 - accuracy: 0.5303
> [0.3139163553714752, 0.5302777886390686]
> Best performing model chosen hyperparameters:
> {'Activation': 'elu', 'Dropout': 0.587606728324542, 'Dropout_1': 0.3746350041674067, 'LSTM': 256, 'kernel_initializer': <tensorflow.python.ops.init_ops_v2.GlorotNormal object at 0x7f622fcd8cc0>, 'loss': <tensorflow.python.keras.losses.MeanSquaredError object at 0x7f622fcd8710>, 'optimizer': <tensorflow.python.keras.optimizer_v2.gradient_descent.SGD object at 0x7f622fcd0898>}
> epochs: 20

Com o seguinte comportamento no gráfico de acurácia e loss obtido no treinamento:

![image](https://user-images.githubusercontent.com/26990166/128136100-ca6197de-fdf6-4366-baf5-d3ab7af247fe.png)

Não foram ótimos resultados, mas o comportamento da curva foi a mais interessante dentre as outras. É tão interessante que o segundo resultado mostrado da matriz a rede teve uma acurácia de treinamento máxima de quase 90% e acurácia de validação de pouco mais de 58%, mas com péssimos resultados em geral.
